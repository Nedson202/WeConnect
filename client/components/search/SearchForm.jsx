import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../../index.scss';

/**
 * @class SearchForm
 * 
 * @extends {Component}
 */
class SearchForm extends Component {
  /**
   * @description Creates an instance of search from.
   * 
   * @param {object} props 
   * 
   * @memberof Navbar
   */
  constructor(props) {
    super(props);

    this.state = {
      query: '',
      option: ''
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * @description Handles input change
   * 
   * @param {any} event
   * 
   * @returns {undefined}
   * 
   * @memberof SearchForm
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value.toLowerCase() });
  }

  /**
   * @description Handles input submission
   * 
   * @param {any} event
   * 
   * @returns {undefined}
   * 
   * @memberof SearchForm
   */
  onSubmit(event) {
    const { query, option } = this.state;
    event.preventDefault();
    this.props.filterBusiness(option, query).then(() => {
      sessionStorage.setItem('query', JSON.stringify(this.state));
      this.context.router.history.push('/searchresult');
    });
  }

  /**
   * @description Renders the component to the dom
   * 
   * @returns {object} JSX object
   * 
   * @memberof BusinessProfile
   */
  render() {
    return (
      <div>
        <form className="form-inline my-2 my-lg-2" onSubmit={this.onSubmit}>
          <input
            className="form-control mr-sm-2"
            type="search"
            placeholder="Filter by location or category"
            name="query"
            onChange={this.onChange}
            value={this.state.query}
            required 
          />
          <select
            className="custom-select"
            name="option"
            onChange={this.onChange}
            value={this.state.option}
            required
          >
            <option value="">choose</option>
            <option value="name">Name</option>
            <option value="location">Location</option>
            <option value="category">Category</option>
          </select>
          <button className="btn btn-outline-success my-2 my-sm-0" type="submit"><i className="fa fa-search" /></button>
        </form>
      </div>
    );
  }
}

SearchForm.contextTypes = {
  router: PropTypes.object.isRequired
};

export default SearchForm;
