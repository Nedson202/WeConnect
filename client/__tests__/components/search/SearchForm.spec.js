import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import SearchForm from '../../../components/search/SearchForm';
import data from '../../__mockData__/userData';

configure({ adapter: new Adapter() });
const { searchState } = data;
let props;
const setup = () => {
  props = {
    state: searchState,
    onChange: jest.fn(),
    onSubmit: jest.fn()
  };

  return shallow(<SearchForm {...props} />);
};


describe('Component: SearchForm', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('select').length).toBe(1);
    expect(wrapper.find('span').length).toBe(0);
    expect(wrapper.find('i').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('.search-button').simulate('click'));
    expect(wrapper.length).toBe(1);
    const state = {
      query: 'query',
      option: 'option',
      isLoading: true
    }
    wrapper.setProps({ state });
    expect(wrapper.find('span').length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  }) 
});