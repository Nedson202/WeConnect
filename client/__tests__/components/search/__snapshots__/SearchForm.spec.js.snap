// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchForm should render 1`] = `
<div>
  <form
    className="form-inline my-2 my-lg-2"
    onSubmit={[MockFunction]}
  >
    <input
      className="form-control mr-sm-2"
      name="query"
      onChange={[MockFunction]}
      placeholder="Filter businesses by"
      type="search"
      value="query"
    />
    <select
      className="custom-select"
      name="option"
      onChange={[MockFunction]}
      value="option"
    >
      <option
        value=""
      >
        choose
      </option>
      <option
        value="name"
      >
        Name
      </option>
      <option
        value="location"
      >
        Location
      </option>
      <option
        value="category"
      >
        Category
      </option>
    </select>
    <button
      className="btn btn-outline-success my-2 my-sm-0 search-button"
      id="search-button"
      type="submit"
    >
      <span>
         searching 
        <i
          className="fa fa-spinner fa-spin"
        />
         
      </span>
    </button>
  </form>
</div>
`;
